### PR TITLE
New Longitudinal Lab Visualizer

### PR DESCRIPTION
**Pull request by ashih2018**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

For JIRA1998.
This is the initial implementation of the Longitudinal Lab Visualizer without interactions.
In the labs section there is a new icon for the lab visualizer which renders our visualizer.
The mock-up given can be seen as:
https://github.com/standardhealth/shr_design/blob/master/Flux%20Notes/screen_designs/shr_fluxnotes_concepts_v17.3.pdf

Note:
This icon currently shows up for vitals. We are working to remove this icon as it is unique for labs. This will need to be addressed before this PR gets merged but we wanted the work to be reviewed.

## Submitter:

**Running the Application**

- [X] Manually tested in Chrome (primary browser for testing)
- [X] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [X] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [X] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [X] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [X] This pull request describes why these changes were made
- [X] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [X] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [X] Code is commented

**Tests**

- [X] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by mgramigna** [comment]
Yay for first PR of many! A couple things of note that will need to be resolved for this visualizer:

[MG1] The visualizer should be a generic component that has no specific references to the section that it is visualizing data for. I know that in this case, our targeted use case is Labs, but that's not to say we won't have need for the longitudinal visualizer for other sections. We should remove all specific variable names and stuff referencing labs, and the components should be viewed as more generic for a way to visualize ValueOverTime data, regardless of whether or not it is lab data

[MG2] I know we talked about disabling this visualizer for vitals, which may still happen. However, we still want this visualizer to work for the vitals section (and potentially other sections in the future). This goes along with the section being more generic and not necessarily geared towards labs. This visualization should theoretically work for both Labs and Vitals without having to change anything. First step for this would be ensuring that the visualizer doesn't break when there are no "bands" in the data. It's possible this might be the only error, but there could be more. This is just the first one I saw

**Comment by mgramigna** [comment]
[MG3] I think we should add the `subsectionLabel` to more than just the Sarcoma metadata. The `/pilotnp` endpoint has both the vitals and labs section, and the `/demo1` and `/demo2` endpoints both have the labs section. On all of these endpoints, the first column doesn't have a header. Here are the relevant files where this should be added:

* `SarcomaNursePractitionerMetadata.jsx` for `/pilot1np`
* `BreastCancerMetadata.jsx` for `/demo1` and `/demo2`

**Comment by gregquinn2001** [comment]
For MG3, @mgramigna  can you show them how to create a new class for VitalsSection and one for LabsSection and then reuse it across the others. They should include the content that's in each of the Metadata's you identify including all the subsection references.

**Comment by mgramigna** [comment]
Yep!

**Comment by gregquinn2001** [comment]
DefaultMetadata and BreastCancerMetadata both have Labs sections as well but with different subsections. We should add the subsectionLabel if there's more than one subsection. DefaultMetadata only has 1 though so let's create a task/subtask for them to add the ability to suppress visualizers. Maybe SuppressVisualizers: [ "visualizername1", "visualizername2", ... ]

**Comment by ashih2018** [comment]
MG3 RESOLVED!!

**Comment by gregquinn2001** [comment]
[GQ1] Let's rename LongitudinalViewVisualizer. View and Visualizer are really the same thing. We need to ask what uniquely defines the class? I think it shows a table where the columns are dates to show values over time so how about LongitudinalTableVisualizer?

**Comment by sgober** [comment]
This is looking great @jpa399 and @ashih2018!

[SO1]  Right now the table doesn't adjust to the size of my screen (see screenshots below).  There is empty white space in the first one and a horizontal overflow when I open a note.  Can we make the table width dynamic to automatically fill all the space?

Screen Shot 2019-06-13 at 11 18 08 AM
Screen Shot 2019-06-13 at 11 18 18 AM

[SO2]  This is picky, but the icon for the visualizer is hard to read when it is selected.  Instead of inverting what is filled, can we just make the color darker like it is done on the other two?

Screen Shot 2019-06-13 at 11 19 56 AM
Screen Shot 2019-06-13 at 11 20 29 AM

File: src/summary/VisualizerManager.jsx [hunk 1]
```diff
@@ -367,4 +371,25 @@ export default class VisualizerManager {
             </svg>
         );
     }
+
+    _longitudinalViewIcon = (isSelected) => {
+        const strokeColor = isSelected ? "#3F3F3F" : "#CCCCCC";
+        const fillColor = isSelected ? "#8d9196" : "#C4C4C4";
```
Comment (sgober) [inline src/summary/VisualizerManager.jsx]: I would make these the same as the stroke color for consistency

**Comment by mgramigna** [comment]
@sgober none of our other tables fill the screen width (procedures, imaging, genetics, ROS, etc)

**Comment by sgober** [comment]
@mgramigna valid, but none of them have a horizontal scroll, so that part of SO1 still stands.  I will edit my comment

**Comment by mgramigna** [comment]
There is a Jira for adding simple time navigation to the table. The horizontal scroll is more or less a placeholder for now. We're thinking of functionality similar to the timeline. Though I guess we could try some investigation for dynamically resizing the table, though it seems it wasn't trivial I'm not sure

File: src/summary/VisualizerManager.jsx (hunk 1 above)
Comment (ashih2018) [reply src/summary/VisualizerManager.jsx]: resolved

**Comment by ashih2018** [comment]
GQ1 Resolved!

**Comment by ashih2018** [comment]
SO2 Resolved!

**Comment by jafeltra** [comment]
I took a look at SO1 and I don't think it makes sense to address as part of this PR. If the table fit the window without the scrollbar, I think it would be too difficult to read. Also, since one of the other tasks is to add the time scroll/scrubber bar, I think the resizing would need to be undone since we'd want the table to scroll in that case. So my opinion would be to not address SO1 now.

**Comment by jpa399** [comment]
We think we've addressed everything now so if commenters could review it again, it should be ready to go! Thanks!

**Comment by sgober** [comment]
Looks good to me!  Just needs a rebase then I will approve & merge!

**Comment by jpa399** [comment]
rebased!

**Comment by sgober** [comment]
waiting on travis...

**Pull request merged**